src/components/profile/profile-settings/Profile-settings.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react'
+
+import { DevTool } from '@hookform/devtools'
+import { zodResolver } from '@hookform/resolvers/zod'
+import Link from 'next/link'
+import { useForm } from 'react-hook-form'
+import { z } from 'zod'
+
+import s from './profileSettings.module.scss'
+
+import { useTranslate } from '@/src/assets/hooks/useTranslate'
+import { CalendarOutline } from '@/src/assets/icons/calendar-outline'
+import { ImgOutline } from '@/src/assets/icons/image-outline'
+import { profileSettingSchema } from '@/src/common/schemas/profile-setting-schema'
+import { Button } from '@/src/components/ui/button'
+import { ControlledTextField } from '@/src/components/ui/controlled'
+import { TextAreaField } from '@/src/components/ui/text-area'
+import { Typography } from '@/src/components/ui/typography'
+
+export type ProfileSettingType = z.infer<typeof profileSettingSchema>
+
+type ProfileSettingPropsType = {
+  onSubmitHandler: (data: ProfileSettingType) => void
+}
+
+export const ProfileSettings = (props: ProfileSettingPropsType) => {
+  const { onSubmitHandler } = props
+  const { t } = useTranslate()
+
+  const { control, handleSubmit } = useForm<ProfileSettingType>({
+    resolver: zodResolver(profileSettingSchema),
```

Comment:
you should add "t" to args to schema , to use translate in zod errors  , so you can see the example in register form

Reply:
done